airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -118,10 +131,18 @@ def __init__(
         self.dynamodb_scan_kwargs = dynamodb_scan_kwargs
         self.s3_bucket_name = s3_bucket_name
         self.s3_key_prefix = s3_key_prefix
-        self.aws_conn_id = aws_conn_id
+        if aws_conn_id is not NOTSET:
+            warnings.warn(_DEPRECATION_MSG, DeprecationWarning, stacklevel=3)
+            self.source_aws_conn_id = aws_conn_id
+        else:
+            self.source_aws_conn_id = source_aws_conn_id
+        if dest_aws_conn_id is NOTSET:
+            self.dest_aws_conn_id = self.source_aws_conn_id
+        else:
+            self.dest_aws_conn_id = dest_aws_conn_id
```

Comment:
"Tests Police" here 🤣 , we need also test this one. Just sure that we cover all cases with different inputs of `aws_conn_id`, `source_aws_conn_id`, `dest_aws_conn_id`, especially (but not limited) cases which is possible for current version of provider:
1. User not provide anything, for example user expected to use `aws_default` connection 🙄 
2. User provide value only to `aws_conn_id`

Might be something like this:

https://github.com/apache/airflow/blob/16fddbae83d03c9b3e2d249cc8852fb006c65c3b/tests/providers/slack/hooks/test_slack_webhook.py#L194-L211

Reply:
To be fair, that wasn't a test for `aws_default` value before 😆 and so far only `test_gcs_to_s3.py` had such test case.
But it's alright, I can add such test and the ones for all the other connection combinations.

Reply:
@Taragolis I need your advise here please. 
None of the other AWS test suites seems to do any assertions on the warning message about deprecation (i.e. `parameter has been deprecated`) (3 other operators are using it: `image_attachment_to_s3`, `s3_to_sftp`, `mongo_to_s3`),
and my code doesn't raise an Exception.

Do you still think it's necessary to assert that the warning is present? If so, why wasn't it done this way in any of those other test suites?

Reply:
That just sample that test could be pretty dumb.
And you could test that warning raised by [`pytest.warn`](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns) context manager it is almost the same as `pytest.raises`

Reply:
Thanks @Taragolis , I added a few more test for most important (all?) remaining combinations.